### ISSUE TITLE
False positive W0108 (unnecessary lambda)

### ISSUE THREAD

**Issue by thehcma** [PyCQA/pylint#1709]
### Steps to reproduce

`lambda x: tuple(x)
` 

### Current behavior
This will be flagged as an unnecessary lambda, but note that, while we are converting something to a tuple, ``tuple`` while appearing to be a function, isn't. This is a type cast.

This comes up in practice, for instance, when using Pandas.

The following does not work (precisely because ``tuple`` is not a function):

`grouped_df = df.groupby(group_by_attributes, as_index=False).aggregate(tuple)
`

The workaround:

`grouped_df = df.groupby(group_by_attributes, as_index=False).aggregate(lambda x: tuple(x))
`

Leads to W0108 in pylint.

### Expected behavior

No warning should be issued, by verifying the type of what is in the lambda. :)

### pylint --version output

pylint 1.7.2, 
astroid 1.5.3
Python 2.7.12 (default, Nov 19 2016, 06:48:10) 
[GCC 5.4.0 20160609]

**Comment by rogalski** [comment]
> `tuple` while appearing to be a function, isn't. This is a type cast.
Is there really a difference in Python? It'll be dispatched to `__call__` eventually anyway. Or is it some weird quick on C-level implementation of Pandas?

**Comment by thehcma** [comment]
You are indeed correct - the dispatching of the functions in ``core/groupby.py`` in the ``aggregate`` function in the ``pandas`` package is, at best, iffy.

While the parameter is called ``func_of_funcs``, it doesn't really dispatch functions.

So we can close this - I will file a bug with the ``pandas`` maintainers.

**Comment by PCManticore** [comment]
Thanks for the issue. Unfortunately this is such an edge case, that simply disabling the message locally next to the line where this happens should do the work. Definitely needs to be addressed by `pandas`!

**Comment by Ivanca** [comment]
One year later and this still an issue...

```python
def first_true_result(seq, predicate, default=None):
    """ gets first callback return that evaluates as tre """
    for i in seq:
        result = predicate(i)
        if result:
            return result
    return default

alias_current_path = first_true_result(aliases, lambda k: aliases[k] if views_module.startswith(k) else None)
```
Pylint cires about "lambda may be unnecessary"
